skpro/regression/linear/_glm.py

```diff
@@ -0,0 +1,443 @@
+
+    pvalues_ : ndarray
+        The two-tailed p-values for the parameters.
+
+    scale_ : float
+        The estimate of the scale / dispersion for the model fit.
+        See GLM.fit and GLM.estimate_scale for more information.
+
+    stand_errors_ : ndarray
+        The standard errors of the fitted GLM.
+    """
+
+    _tags = {
+        "authors": ["julian-fong"],
+        "maintainers": ["julian-fong"],
+        "python_version": None,
+        "python_dependencies": "statsmodels",
+        "capability:multioutput": False,
+        "capability:missing": False,
+        "X_inner_mtype": "pd_DataFrame_Table",
+        "y_inner_mtype": "pd_DataFrame_Table",
+    }
+
+    def __init__(
+        self,
+        offset=None,
+        exposure=None,
+        freq_weights=None,
+        var_weights=None,
+        missing="none",
+        start_params=None,
+        maxiter=100,
+        method="IRLS",
+        tol=1e-8,
+        scale=None,
+        cov_type="nonrobust",
+        cov_kwds=None,
+        use_t=None,
+        full_output=True,
+        disp=False,
+        max_start_irls=3,
+        add_constant=False,
+    ):
+        super().__init__()
+        from statsmodels.genmod.families.family import Gaussian
+
+        self.family = Gaussian()
+        self.offset = offset
+        self.exposure = exposure
+        self.freq_weights = freq_weights
+        self.var_weights = var_weights
+        self.missing = missing
+        self.start_params = start_params
+        self.maxiter = maxiter
+        self.method = method
+        self.tol = tol
+        self.scale = scale
+        self.cov_type = cov_type
+        self.cov_kwds = cov_kwds
+        self.use_t = use_t
+        self.full_output = full_output
+        self.disp = disp
+        self.max_start_irls = max_start_irls
+        self.add_constant = add_constant
+
+    def _fit(self, X, y):
+        """Fit regressor to training data.
+
+        Writes to self:
+            Sets fitted model attributes ending in "_".
+
+        Parameters
+        ----------
+        X : pandas DataFrame
+            A (n,k) array where n is the number of observations and k is the number
+            of regressors. An intercept is not included by default and should be
+            added by the user (models specified using a formula include an
+            intercept by default). Equivalent to statsmodel's (exog).
+
+        y : pandas DataFrame
+            1d array of the endogenous response variable. This array can be 1d or
+            2d. Binomial family models accept a 2d array  with two columns.
+            If supplied each observation is expected to  be [success, failure].
+            Equivalent to statsmodel's (endog).
+
+        Returns
+        -------
+        self : reference to self
+        """
+        from statsmodels.genmod.generalized_linear_model import GLM
+        from statsmodels.tools import add_constant
+
+        if self.add_constant:
```

Comment:
do we need to do this in `_fit`, too?

Reply:
yes, i'll change the code so its consistent with the other methods

Reply:
I thought `statsmodels` does it in `fit`?

Reply:
As per my knowledge, the ```GLM``` model on `statsmodels` does not automatically include a constant column at any point during any fitting or prediction process

Reply:
I see, so you added this feature?